### PR TITLE
Only `parallelize(for_generation=True)` if pipeline task is in `SUPPORTED_GENERATION_TASKS`

### PR DESCRIPTION
**Pull request by HMellor**
# What does this PR do?

Fixes failing pipeline tests coming from index error in `fill-mask` pipeline

To validate:
```
RUN_SLOW=true pytest -k BartForConditionalGeneration tests/pipelines/test_pipelines_fill_mask.py -v
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/pipelines/test_pipelines_fill_mask.py [hunk 1]
```diff
@@ -202,7 +198,7 @@ def run_pipeline_test(self, fill_masker, examples):
             ],
         )
 
-        outputs = fill_masker([f"This is a {tokenizer.mask_token}", f"Another {tokenizer.mask_token} great test."])
+        outputs = fill_masker(examples * 2)
```
Comment (katalinic-gc) [inline tests/pipelines/test_pipelines_fill_mask.py:201]: How come you are changing the test example here?

Comment (HMellor) [reply tests/pipelines/test_pipelines_fill_mask.py:201]: Because currently this test is not using the `examples` provided by the function arguments. So I just changed the test cases so that they did. It's still testing `str, list(str), list(str, str)` like it was before.

Comment (katalinic-gc) [reply tests/pipelines/test_pipelines_fill_mask.py:201]: Do you by any chance know when the test started failing? We didn't do a transformers update so the need to add postprocess seems like it would have been failing for quite a while 🤔

Comment (HMellor) [reply tests/pipelines/test_pipelines_fill_mask.py:201]: You're right that this is probably the wrong location for the fix.

The issue is that the logits output from BART are of shape `[batch_size, 1, vocab_size]` (presumably the 1 is the number of mask tokens) but for other models like BERT the logits are of shape `[batch_size, sequence_length, vicab_size]`.

The added `postprocess` tweak works because for all other models the `expand` is a no-op, but for BART it corrects the output shape (by faking all the other logits). 

To truly solve this we need to make sure that BART outputs the logits for the full sequence.

**Review by katalinic-gc — APPROVED** [review]
